emcc.py

```diff
@@ -394,18 +394,31 @@ def find_output_arg(args):
   """
   outargs = []
   specified_target = None
-  use_next = False
-  for arg in args:
-    if use_next:
-      specified_target = arg
-      use_next = False
-      continue
+
+  arg_count = len(args)
+  arg_index = 0
+
+  while arg_index < arg_count:
+    arg = args[arg_index]
+
     if arg == '-o':
-      use_next = True
-    elif arg.startswith('-o'):
+      specified_target = args[arg_index + 1]
```

Comment:
Probably should check for `arg_index < arg_count - 1` here too (seems a little safer than doing `arg_index != arg_count - 1`).

Do you find this new version simpler?  Its not a clear win to me but I don't feel strongly about it.  Its certainly more code.

lgtm with the above fix if you feel that its easier to understand.

Reply:
I added the check. I used the `!=` version because there is no reason for going out of bounds here. If it is the case it's a bug and I prefer the program to fail to we can see it.

> Do you find this new version simpler? Its not a clear win to me but I don't feel strongly about it. Its certainly more code.

It's indeed more code, t's also not a very Python way to write a loop :) I personally think it's easier to reason about as each iteration is self contained, with no state variable. In the previous version I had to process the loop in my mind to get it (_if `use_next` is true it means that we had `-o` in the previous iteration but maybe we have `skip = True` too? and so if I have `skip = True` I must set it to false otherwise it will skip in the next iteration and aaaaaaaargh`_).